pkg/kubectl/cmd/delete.go

```diff
@@ -167,6 +173,9 @@ func (o *DeleteOptions) Complete(f cmdutil.Factory, args []string, cmd *cobra.Co
 		o.WaitForDeletion = true
 		o.GracePeriod = 1
 	}
+	if b, err := cmd.Flags().GetBool("wait"); err == nil {
+		o.WaitForDeletion = b
```

Comment:
Actually, once we add an explicit `--wait` flag and set it to `true` by default, I don't think we need to keep the existing logic for backward compatibility at all?

I mean the condition https://github.com/kubernetes/kubernetes/blob/0db40da90975eef37b35e2c859441703f351168e/pkg/kubectl/cmd/delete.go#L163-L168 (and 2 other similar places).
With the changes in this PR, `WaitForDeletion` is already `true` by default, i.e. it's consistent with the backward compatibility for grace period. And if the user explicitly sets `--wait=false`, we shouldn't override it there, I think?

Reply:
To be clear about the change I propose: Get rid of the existing logic with conditions for setting internal `WaitForDeletion` to `true` completely, and declare a new `--wait` flag that is exposed to the user in the same way as other flags in `deleteFlags.AddFlags(...)`.

Reply:
Once reapers die, perhaps.

Reply:
Yes, that's the goal of getting rid of reapers. The old logic will die.

Reply:
> So this seems to be the reason for defining the "wait" flag differently from the rest - reusing the existing WaitForDeletion flag that is currently not exposed to the user?

I think it falls out in the other direction.  Reapers die and we remove this old codepath.

Reply:
Once the reapers are dead, we can move the wait flag to `deleteFlags.AddFlags(...)` to make it consistent with other flags, since the default will always be `true`, and there won't be internal conditional "graceful deletion" anymore.